### PR TITLE
feat(Response): add response.remoteAddress method

### LINKED ISSUE #2171: Can I get Remote Address with puppteer？
Can I get Remote Address with puppteer？

### PR DESCRIPTION
**Pull request by szdc**
This patch exposes a remoteAddress property which is a formatted string containing the IP address and port of the remote server

Closes #2171 

I wasn't sure if I should expose the remote address and port separately (rather than a formatted version), so please let me know if that's a better idea.

### CODE REVIEW

**Comment by szdc** [comment]
Thoughts on how I tackle the IPv4 loopback address in the Travis build?

File: lib/NetworkManager.js [hunk 1]
```diff
@@ -188,7 +188,7 @@ class NetworkManager extends EventEmitter {
    * @param {?Object} securityDetails
    */
   _handleRequestRedirect(request, redirectStatus, redirectHeaders, fromDiskCache, fromServiceWorker, securityDetails) {
-    const response = new Response(this._client, request, redirectStatus, redirectHeaders, fromDiskCache, fromServiceWorker, securityDetails);
+    const response = new Response(this._client, request, redirectStatus, redirectHeaders, fromDiskCache, fromServiceWorker, securityDetails, null, null);
```
Comment (aslushnikov) [inline lib/NetworkManager.js]: let's plumb the remoteIDAddress and remotePort to the redirect responses (also let's cover this with a test)

File: examples/search.js [hunk 2]
```diff
@@ -27,8 +27,6 @@ const puppeteer = require('puppeteer');
   const browser = await puppeteer.launch();
   const page = await browser.newPage();
 
-  await page.goto('https://developers.google.com/web/');
```
Comment (aslushnikov) [inline examples/search.js]: why this change?

File: docs/api.md [hunk 3]
```diff
@@ -2513,6 +2514,9 @@ This method will throw if the response body is not parsable via `JSON.parse`.
 
 Contains a boolean stating whether the response was successful (status in the range 200-299) or not.
 
+#### response.remoteAddress()
+- returns: <[string]> The formatted address and port of the remote server.
```
Comment (aslushnikov) [inline docs/api.md]: I'd return IP address and port separately: it's a little error-prone to parse out port number with IPv6 addresses.

For example, this method can return an object `{ip, port}`

File: test/network.spec.js [hunk 4]
```diff
@@ -56,6 +56,7 @@ module.exports.addTests = function({testRunner, expect}) {
       expect(responses[0].fromCache()).toBe(false);
       expect(responses[0].fromServiceWorker()).toBe(false);
       expect(responses[0].request()).toBeTruthy();
+      expect(responses[0].remoteAddress()).toBe(`[::1]:${server.PORT}`);
```
Comment (aslushnikov) [inline test/network.spec.js]: I'd try navigating to IP explicitly in line 51 to make this less flaky across environments (different setups might resolve
`localhost` differently)

```js
await page.goto(`http://[::1]:${server.PORT}/empty.html`);
```

File: examples/search.js (hunk 2 above)
Comment (szdc) [reply examples/search.js]: Accidental; sorry.

File: lib/NetworkManager.js (hunk 1 above)
Comment (szdc) [reply lib/NetworkManager.js]: Good point - I'll update with a test later today.

File: test/network.spec.js (hunk 4 above)
Comment (szdc) [reply test/network.spec.js]: We'll still have trouble on IPv4-only machines as they won't be able to handle `[::1]`.  If we use `127.0.0.1` instead, Chrome still returns `[::1]` on IPv6-capable machines.

**Comment by aslushnikov** [comment]
Close in favor of #3192.
